driver-core/src/main/java/com/datastax/driver/core/ControlConnection.java

```diff
@@ -472,12 +479,30 @@ private static void updateInfo(Host host, Row row, Cluster.Manager cluster, bool
             updateLocationInfo(host, row.getString("data_center"), row.getString("rack"), isInitialConnection, cluster);
 
         String version = row.getString("release_version");
-        // We don't know if it's a 'local' or a 'peers' row, and only 'peers' rows have the 'peer' field.
-        InetAddress listenAddress = row.getColumnDefinitions().contains("peer")
-                ? row.getInet("peer")
+        host.setVersion(version);
+
+        // Before CASSANDRA-9436 local row did not contain any info about the host addresses.
+        // After CASSANDRA-9436 (2.0.16, 2.1.6, 2.2.0 rc1) local row contains two new columns:
+        // - broadcast_address
+        // - rpc_address
+        // After CASSANDRA-9603 (2.0.17, 2.1.8, 2.2.0 rc2) local row contains one more column:
+        // - listen_address
+
+        InetAddress broadcastAddress = null;
+        if (row.getColumnDefinitions().contains("peer")) { // system.peers
+            broadcastAddress = row.getInet("peer");
+        } else if (row.getColumnDefinitions().contains("broadcast_address")) { // system.local
+            broadcastAddress = row.getInet("broadcast_address");
+        }
+        host.setBroadcastAddress(broadcastAddress);
+
+        // in system.local only for C* versions >= 2.0.17, 2.1.8, 2.2.0 rc2,
+        // not yet in system.peers as of C* 3.2
+        InetAddress listenAddress = row.getColumnDefinitions().contains("listen_address")
```

Comment:
I'll add a test separate of this PR to ensure that if the peers table contains listen address that it's properly parsed or set  (can mock that up with scassandra).